orttraining/orttraining/eager/ort_aten.cpp

```diff
@@ -665,18 +665,24 @@ at::Tensor& copy_(
   auto ort_self = create_ort_value(invoker, self);
   if (self.scalar_type() != src.scalar_type()) {
     // invoke cast first
-    std::vector<OrtValue> ort_cast_output(1);
-    onnxruntime::NodeAttributes attrs(1);
-    attrs["to"] = create_ort_attribute(
-        "to", (int64_t)GetONNXTensorProtoDataType(self.scalar_type()), at::kLong);
-
-    auto status = invoker.Invoke("Cast",
-                                 {std::move(ort_src)},
-                                 ort_cast_output, &attrs);
-
-    CHECK_STATUS(status);
-
-    copy(invoker, ort_cast_output[0], ort_self);
+    if(self.device().type() != src.device().type()){
+      auto val = at::native::to(src, self.scalar_type());
+      copy_(self, val, false);
+    }else{
+      //Will we ever encounter a case when Scalar type is different but device type is the same?
+      std::vector<OrtValue> ort_cast_output(1);
```

Comment:
you don't need to invoke ort cast manually, just reuse the aten::to to let it dispatch to ort

Reply:
Updated code as per offline discussion. Latest code changes are validated locally for the specific failure case mentioned in the PR.